test/support/utils.rb

```diff
@@ -91,6 +89,22 @@ def create_pseudo_terminal(boot_options: "-r debug/run")
 
     private
 
+    # use this to start a debug session with the test program
+    def start_debugger_with(program)
```

Comment:
It is difficult to know the feature from debug log, so could you try to make clearer name for this feature?

Reply:
do you think `_start_manual_debug_session` is a good name? I put an underscore to indicate that it's a helper.

Reply:
how about `manual_debug_code`?